diff --git a/kernel/signal.c b/kernel/signal.c
index 823d7603b..24eada104 100755
--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -51,6 +51,9 @@
 #include <asm/unistd.h>
 #include <asm/siginfo.h>
 #include <asm/cacheflush.h>
+#ifdef CONFIG_REKERNEL
+#include <../drivers/rekernel/rekernel.h>
+#endif /* CONFIG_REKERNEL */
 #include "audit.h"	/* audit_signal_info() */
 #ifdef OPLUS_FEATURE_HANS_FREEZE
 // Kun.Zhou@AD.RESCONTROL, 2019/09/23, add for hans freeze manager
@@ -1239,6 +1242,11 @@ int do_send_sig_info(int sig, struct siginfo *info, struct task_struct *p,
 	unsigned long flags;
 	int ret = -ESRCH;
 
+#ifdef CONFIG_REKERNEL
+	if (p && (sig == SIGKILL || sig == SIGTERM || sig == SIGABRT || sig == SIGQUIT))
+		rekernel_report(REKERNEL_SIGNAL, sig, current->pid, current,
+				task_pid_nr(p), p, 0, NULL);
+#endif /* CONFIG_REKERNEL */
 #ifdef OPLUS_FEATURE_HANS_FREEZE
 //#Kun.Zhou@ANDROID.RESCONTROL, 2019/09/23, add for hans freeze manager
 	if (is_frozen_tg(p)  /*signal receiver thread group is frozen?*/
