diff --git a/drivers/android/binder.c b/drivers/android/binder.c
index 6a8ec6e..808bf53 100644
--- a/drivers/android/binder.c
+++ b/drivers/android/binder.c
@@ -54,6 +54,9 @@
 #include <linux/fdtable.h>
 #include <linux/file.h>
 #include <linux/freezer.h>
+#ifdef CONFIG_REKERNEL
+#include "../rekernel/rekernel.h"
+#endif /* CONFIG_REKERNEL */
 #include <linux/fs.h>
 #include <linux/list.h>
 #include <linux/miscdevice.h>
@@ -3068,6 +3071,27 @@ static void binder_transaction(struct binder_proc *proc,
 		target_proc = target_thread->proc;
 		target_proc->tmp_ref++;
 		binder_inner_proc_unlock(target_thread->proc);
+#ifdef CONFIG_REKERNEL
+		if (target_proc && target_proc->tsk) {
+			if (reply)
+				binder_reply_handler(proc->pid, proc->tsk,
+						     target_proc->pid,
+						     target_proc->tsk, 0, tr);
+			else {
+				binder_trans_handler(proc->pid, proc->tsk,
+						     target_proc->pid,
+						     target_proc->tsk,
+						     tr->flags & TF_ONE_WAY, tr);
+				if ((tr->flags & TF_ONE_WAY) &&
+				    target_proc->alloc.free_async_space <
+					(target_proc->alloc.buffer_size / 10 + 0x300))
+					binder_overflow_handler(proc->pid, proc->tsk,
+								target_proc->pid,
+								target_proc->tsk,
+								true, tr);
+			}
+		}
+#endif /* CONFIG_REKERNEL */
 	} else {
 		if (tr->target.handle) {
 			struct binder_ref *ref;
